enhancements/cluster-scope-secret-volumes/shared-resources-via-openshift-builds-operator.md

```diff
@@ -0,0 +1,291 @@
+---
+title: shared-resources-via-openshift-builds-operator
+authors:
+  - "@adambkaplan"
+  - "@gabemontero"
+reviewers: # Include a comment about what domain expertise a reviewer is expected to bring and what area of the enhancement you expect them to focus on. For example: - "@networkguru, for networking aspects, please look at IP bootstrapping aspect"
+  - "@cdaley"
+  - "@rgormley"
+approvers: # A single approver is preferred, the role of the approver is to raise important questions, help ensure the enhancement receives reviews from all applicable areas/SMEs, and determine when consensus is achieved such that the EP can move forward to implementation.  Having multiple approvers makes it difficult to determine who is responsible for the actual approval.
+  - "@bparees"
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - "@bparees"
+creation-date: 2023-08-17
+last-updated: 2023-08-22
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - https://issues.redhat.com/browse/RHDP-701
+see-also:
+  - "enhancements/cluster-scope-secret-volumes/csi-driver-host-injections.md"
+  - "enhancements/cluster-scope-secret-volumes/shared-resource-validation.md"
+replaces:
+  - "/enhancements/subscription-content/subscription-injection.md"
+superseded-by: []
+---
+
+# Shared Resources via OpenShift Builds Operator
+
+## Summary
+
+Remove the Shared Resources CSI driver from OpenShift, and deliver it to customers with the
+forthcoming OpenShift Builds OLM Operator.
+
+
+## Motivation
+
+The Shared Resources CSI driver provides a novel way for sharing information across OpenShift
+namespaces. Though there was hope this capability would be generally useful, in practice this
+feature is mainly needed for Red Hat build workloads.
```

Comment:
we've asked mgmt/staffing engineers quite a number of times since its tech preview introduction back in 4.7 about actually plans for non build use cases and have been told "no plans" and that this thing is really only a "build" thing.  

Also, according to PMs, all the 20 some odd external customers listed in OCPSTRAT-201 are characterized as only interested for builds.

So given finite resource constraints, we are admittedly focused on sorting out the majority use case vs. others which have been suggested, but never got prioritized.  We are past the point of accommodating "might be nice". The time betwen 4.7 and 4.14 seems like a sufficient runway to see if these non-build use cases would get prioritized.

The harsh reality as well is that with the old "developer experience" team no longer part of core OCP for a couple of years now,  but still the owner of shared resources and builds (where attempts to transfer shared resource ownership to within OCP have been rejected), with each passing release cycle, we are more and more out of touch with OCP processes, schedules, fine grained technical details, among others.  Shipping shared resources in the same way we ship all our other deliverables bypasses a lot of "indirect" costs outside of the actual coding/testing/etc.